### PR TITLE
OSX publish gems script

### PR DESCRIPTION
**Pull request by sapieneptus**
```
make publish_osx
```

Installs most recent gems, zips, and pushes to S3. OS X Only 

*Requires `awe-cli` installed and configured (`brew install awscli`). Get creds from Kasper or Karl. 

@jonstoneman Can we coordinate on getting a parallel script up for Windows?

### CODE REVIEW

**Comment by sapieneptus** [comment]
@jmoody Any idea why travis is failing? It seems like it all goes well, then suddenly `exit 127`

**Comment by jmoody** [comment]
I don't see why it is failing.  This output is from Jenkins:

```
03:42:13 override rwxr-xr-x  moody/staff for /usr/local/bin/calabash-sandbox? (y/n [n]) not overwritten
03:42:13 Done! Installed:
03:42:13 calabash-ios:       bash: line 1: calabash-ios: command not found
03:42:13 calabash-android:   bash: line 1: calabash-android: command not found
03:42:13 xamarin-test-cloud: bash: line 1: test-cloud: command not found
03:42:13 Execute 'calabash-sandbox update' to check for gem updates.
03:42:13 Execute 'calabash-sandbox' to get started! 
```

**Comment by sapieneptus** [comment]
@jmoody Seems to be working now. Can you review bin/publish/osx.sh ?

**Comment by krukow** [comment]
@jmoody are we good to merge this?

**Comment by jmoody** [comment]
I need to review.

**Comment by sapieneptus** [comment]
FWIW the latest sandbox gem zip was created and uploaded using this script.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+#
```
Comment (jmoody) [inline Makefile]: +1

File: bin/publish/osx.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+function banner {
```
Comment (jmoody) [inline bin/publish/osx.sh]: Nice.

File: bin/publish/osx.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  echo "$(tput setaf 2)INFO: $1$(tput sgr0)"
+}
+
+function error {
+  echo "$(tput setaf 1)ERROR: $1$(tput sgr0)"
+  exit $2
+}
+
+#usage: validate_cmd <exit_code> <action_message> <error_code_if_unsuccessful>
+function validate_cmd {
```
Comment (jmoody) [inline bin/publish/osx.sh]: Even nicer.

**Comment by jmoody** [comment]
Sweet!

I made a PR to fix up some issues.

Back to you @sapieneptus - merge as you see fit.

Thanks!

File: bin/publish/windows.bat [hunk 4]
```diff
@@ -0,0 +1 @@
+#TODO
```
Comment (sapieneptus) [inline bin/publish/windows.bat]: @jonstoneman At some point we should talk about this.

**Comment by sapieneptus** [comment]
I see fit.

**Comment by jmoody** [comment]
Ack!

I was hoping you would merge this

**Feature/automate sandbox bundling - jmoody edits** #54

into _this_ pull request.

@sapieneptus

**Pull request merged**